### ISSUE TITLE
CUDA jax.random.split fails on Jetson Xavier NX with GPU

### ISSUE THREAD

**Issue by kylestach** [google/jax#14546]
### Description
I obviously can't use the prebuilt wheel files, because Tegra is arm/aarch64 rather than x86_64. `jaxlib` is built from scratch on v0.3.22 with `python3 build/build.py --enable_cuda` on a virtualenv with python 3.8.10.
```python
import jax
jax.random.split(jax.random.PRNGKey(0))
```
Output:
```
2023-02-16 16:53:37.523421: E external/org_tensorflow/tensorflow/compiler/xla/pjrt/pjrt_stream_executor_client.cc:2153] Execution of replica 0 failed: INTERNAL: CustomCall failed: jaxlib/cuda/cuda_prng_kernels.cc:32: operation cudaGetLastError() failed: no kernel image is available for execution on the device
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/USERNAME/jax/jax/_src/random.py", line 213, in split
    return _return_prng_keys(wrapped, _split(key, num))
  File "/home/USERNAME/jax/jax/_src/random.py", line 199, in _split
    return prng.random_split(key, count=num)
  File "/home/USERNAME/jax/jax/_src/prng.py", line 624, in random_split
    return random_split_p.bind(keys, count=count)
  File "/home/USERNAME/jax/jax/_src/prng.py", line 636, in random_split_impl
    base_arr = random_split_impl_base(
  File "/home/USERNAME/jax/jax/_src/prng.py", line 642, in random_split_impl_base
    return split(base_arr)
  File "/home/USERNAME/jax/jax/_src/prng.py", line 641, in <lambda>
    split = iterated_vmap_unary(keys_ndim, lambda k: impl.split(k, count))
  File "/home/USERNAME/jax/jax/_src/prng.py", line 1033, in threefry_split
    return _threefry_split(key, int(num))  # type: ignore
  File "/home/USERNAME/jax/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/home/USERNAME/jax/jax/_src/api.py", line 606, in cache_miss
    out_flat = xla.xla_call(
  File "/home/USERNAME/jax/jax/core.py", line 1939, in bind
    return call_bind(self, fun, *args, **params)
  File "/home/USERNAME/jax/jax/core.py", line 1955, in call_bind
    outs = top_trace.process_call(primitive, fun_, tracers, params)
  File "/home/USERNAME/jax/jax/core.py", line 701, in process_call
    return primitive.impl(f, *tracers, **params)
  File "/home/USERNAME/jax/jax/_src/dispatch.py", line 237, in _xla_call_impl
    return compiled_fun(*args)
  File "/home/USERNAME/jax/jax/_src/dispatch.py", line 878, in _execute_compiled
    out_flat = compiled.execute(in_flat)
jax._src.traceback_util.UnfilteredStackTrace: jaxlib.xla_extension.XlaRuntimeError: INTERNAL: CustomCall failed: jaxlib/cuda/cuda_prng_kernels.cc:32: operation cudaGetLastError() failed: no kernel image is available for execution on the device

The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.

--------------------

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/USERNAME/jax/jax/_src/random.py", line 213, in split
    return _return_prng_keys(wrapped, _split(key, num))
  File "/home/USERNAME/jax/jax/_src/random.py", line 199, in _split
    return prng.random_split(key, count=num)
  File "/home/USERNAME/jax/jax/_src/prng.py", line 624, in random_split
    return random_split_p.bind(keys, count=count)
  File "/home/USERNAME/jax/jax/core.py", line 328, in bind
    return self.bind_with_trace(find_top_trace(args), args, params)
  File "/home/USERNAME/jax/jax/core.py", line 331, in bind_with_trace
    out = trace.process_primitive(self, map(trace.full_raise, args), params)
  File "/home/USERNAME/jax/jax/core.py", line 698, in process_primitive
    return primitive.impl(*tracers, **params)
  File "/home/USERNAME/jax/jax/_src/prng.py", line 636, in random_split_impl
    base_arr = random_split_impl_base(
  File "/home/USERNAME/jax/jax/_src/prng.py", line 642, in random_split_impl_base
    return split(base_arr)
  File "/home/USERNAME/jax/jax/_src/prng.py", line 641, in <lambda>
    split = iterated_vmap_unary(keys_ndim, lambda k: impl.split(k, count))
  File "/home/USERNAME/jax/jax/_src/prng.py", line 1033, in threefry_split
    return _threefry_split(key, int(num))  # type: ignore
jaxlib.xla_extension.XlaRuntimeError: INTERNAL: CustomCall failed: jaxlib/cuda/cuda_prng_kernels.cc:32: operation cudaGetLastError() failed: no kernel image is available for execution on the device
```

### What jax/jaxlib version are you using?

jax v0.3.22, jaxlib v0.3.22 (tried 0.4 as well)

### Which accelerator(s) are you using?

GPU (Tegra)

### Additional system info

Python versions tried: 3.8, 3.9. Linux kernel version: `5.10.104-tegra`.

### NVIDIA GPU info

No `nvidia-smi` on Tegra, but `nvcc --version` gives:
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_May__4_00:02:26_PDT_2022
Cuda compilation tools, release 11.4, V11.4.239
Build cuda_11.4.r11.4/compiler.31294910_0
```

**Comment by azh2000xy** [comment]
have you solved this problem?

**Comment by hawkinsp** [comment]
Normally this means that the jaxlib wheel wasn't built with the right CUDA compute capabilities (see options to `build.py` when building the wheel). However, AIUI, this particular GPU is a Volta variant and Volta support should be included by default. So I'm not sure what's going on.

**Comment by kylestach** [comment]
We have not found a solution to the problem at this point. I've tried explicitly setting the CUDA capability (and CUDNN version) to the device capability when building jaxlib, but no luck.

**Comment by kylestach** [comment]
Update: JAX+CUDA appears to work on the Orin AGX (checkout from v0.4.6). Unfortunately I don't have an Orin NX handy to check if this happens across the whole Orin lineup, but if anyone has one lying around and could test this I'd appreciate it.

**Comment by hzheng40** [comment]
I've managed to make at least the specific call in this issue work on Jetpack 5.1, Xavier AGX.
I checked out the jax-v0.4.6 tag, with a virtual env of python 3.9, and the following call when calling `build.py`:
`python3 build/build.py --enable_cuda --cuda_compute_capabilities=sm_72`. I imagine switching to the specific compute capabilities code to your specific jetson should work.
I'll do further testing and report back if there are any other issues.
